Resources/doc/reference/configuration/user-context.rst

```diff
@@ -120,6 +120,16 @@ request. However, when you decide to cache hash responses, you must invalidate
 them when the user context changes, particularly when the user logs in or out.
 This bundle provides a logout handler that takes care of this for you.
 
+.. _always_vary_on_context_hash:
+
+``always_vary_on_context_hash``
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+**type**: ``boolean`` **default**: ``true``
+
+Set ``always_vary_on_context_hash`` to  ``false`` to disable always adding X-User-Context-Hash or Cookie, Authorization in Vary header.
```

Comment:
You are totally right, we need to Vary on cookie, but only for the response to the hash lookup request, and not for all responses.
It seems to me that this Vary is already set in the kernelRequest with the following code : 
`$response->setVary($this->userIdentifierHeaders);`
so there should be no need to set it again in kernelReponse.
If @lolautruche could indeed confirm that it would be great !